File: go/internal/router.go
package internal

import "net/http"

const (
	HealthEndpoint     = "/health"
	FizzBuzzEndpoint   = "/fizzbuzz"
	StatisticsEndpoint = "/statistics"
)

func NewRouter(handler Handler) *http.ServeMux {
	mux := http.NewServeMux()

	mux.HandleFunc(HealthEndpoint, handler.Health)
	mux.HandleFunc(FizzBuzzEndpoint, handler.FizzBuzz)
	mux.HandleFunc(StatisticsEndpoint, handler.Statistics)

	return mux
}

File: python/internal/router.py
from litestar import Router

from .handler import get_fizzbuzz, get_health, get_statistics

api_router = Router(
    path="",
    route_handlers=(
        Router(
            path="/health",
            route_handlers=(get_health,),
        ),
        Router(
            path="/fizzbuzz",
            route_handlers=(get_fizzbuzz,),
        ),
        Router(
            path="/statistics",
            route_handlers=(get_statistics,),
        ),
    ),
)
